### PR TITLE
test: chain: unit tests for the syncer & sync manager

### PR DESCRIPTION
**Pull request by TheDivic**
Added a few tests for some uncovered methods of the syncer and sync manager. Deleted some dead code.

## Proposed Changes
- Linter reports `sync.go:getLatestBeaconEntry` to be dead code. I also checked for any references (didn't have to look far since it's a private method) and there aren't any, so I deleted it. 
- Wrote some unit tests for the syncer similar to the ones that were already there (in `sync.go`) but I covered the last few methods that didn't have any unit tests.
- Wrote a basic "lifecycle" test (create->insert->pop) for the sync manager bucket set. I know testing private functions is not the best practice but this one implements some important syncer behavior (primarily choosing what to pop).

## Additional Info
- Was `sync.go:getLatestBeaconEntry` left there on purpose? ⚠️

## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [x] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [x] This PR has tests for new functionality or change in behaviour
- [x] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] CI is green

### CODE REVIEW

**Comment by TheDivic** [comment]
@ZenGround0 @magik6k Hey guy I have two test suites that are completely unrelated (or at least I think so) to my changes that are failing: test-itest-ccupgrade and test-itest-paych_api.
Are these tests flaky or I produced some unintended side effects?

**Comment by ZenGround0** [comment]
@TheDivic thanks for the heads up.  Looks like they are flaky.  itest-ccupgrade should be salvageable, I'll look into it.  paych_api I have less confidence in quick turn around but I'll also look.

File: chain/sync_test.go [hunk 1]
```diff
@@ -1098,3 +1098,158 @@ func TestInvalidHeight(t *testing.T) {
...
+	// - after unmarking blocks as bad and reconnecting, source & client should be in sync
+	//stm: @CHAIN_SYNCER_UNMARK_BAD_001
+	err = tu.nds[client].SyncUnmarkBad(tu.ctx, sourceHead.Cids()[0])
+	require.NoError(tu.t, err)
+
+	reason, err = tu.nds[client].SyncCheckBad(tu.ctx, sourceHead.Cids()[0])
+	require.NoError(tu.t, err)
+	require.Equal(tu.t, "", reason, "block is still bad after manually unmarking")
+
+	err = tu.nds[client].SyncUnmarkAllBad(tu.ctx)
```
Comment (ZenGround0) [inline chain/sync_test.go:1193]: I'm new to the lotus syncer so I'm curious: why do we need the explicit `SyncUnmarkBad` on the head if we are unmarking all here?

**Review by ZenGround0 — APPROVED** [review]
LGTM.  We should wait for @arajasek or @magik6k to sign off on removal of `getLatestBeaconEntry`.

File: chain/sync_test.go (hunk 1 above)
Comment (brdji) [reply chain/sync_test.go:1193]: I believe it's done intentionally, so as to check that both methods of unmarking a bad block are working as intended. In this way, if a change is commited to either `SyncUnmarkBad` or `SyncUnmarkAllBad`, the test will cover them.

While this test strays into the "integration" category, I think it's a valid choice.

File: chain/sync_manager_test.go [hunk 2]
```diff
@@ -240,3 +241,37 @@ func TestSyncManager(t *testing.T) {
 		op3.done()
 	})
 }
+
+func TestSyncManagerBucketSet(t *testing.T) {
+	ts1 := mock.TipSet(mock.MkBlock(nil, 0, 0))
+	ts2 := mock.TipSet(mock.MkBlock(ts1, 1, 0))
+	bucket1 := newSyncTargetBucket(ts1, ts2)
+	bucketSet := syncBucketSet{buckets: []*syncTargetBucket{bucket1}}
+	fmt.Println("bucketSet: ", bucketSet.String())
```
[the remark this answers is not in the document]
Comment (brdji) [reply chain/sync_manager_test.go:250]: Removed.

File: chain/sync.go [hunk 3]
```diff
@@ -1244,25 +1244,3 @@ func (syncer *Syncer) CheckBadBlockCache(blk cid.Cid) (string, bool) {
 	bbr, ok := syncer.bad.Has(blk)
 	return bbr.String(), ok
 }
-
-func (syncer *Syncer) getLatestBeaconEntry(ctx context.Context, ts *types.TipSet) (*types.BeaconEntry, error) {
```
[the remark this answers is not in the document]
Comment (ZenGround0) [reply chain/sync.go:1248]: Ping @arajasek

**Review by arajasek — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**